spec/acceptance/018_shield_spec.rb

```diff
@@ -0,0 +1,306 @@
+require 'spec_helper_acceptance'
+
+describe "elasticsearch shield" do
+
+  # Template manifest
+  let :base_manifest do <<-EOF
+class { 'elasticsearch' :
+  java_install => true,
+  manage_repo  => true,
+  repo_version => '#{test_settings['repo_version']}',
+}
+
+elasticsearch::plugin { 'elasticsearch/license/latest' :  }
+elasticsearch::plugin { 'elasticsearch/shield/latest' : }
+EOF
+  end
+
+  describe 'user authentication' do
+
+    describe 'single instance manifest' do
+
+      let :single_manifest do
+        base_manifest + <<EOF
+elasticsearch::instance { ['es-01'] :  }
+
+Elasticsearch::Plugin { instances => ['es-01'],  }
+
+elasticsearch::shield::user { '#{@user}':
+  password => '#{@user_password}',
+  roles    => ['admin'],
+}
+elasticsearch::shield::user { '#{@user}pwchange':
+  password => '#{@user_password}',
+  roles    => ['admin'],
+}
+EOF
+      end
+
+      it 'should apply cleanly' do
+        apply_manifest single_manifest, :catch_failures => true
+      end
+
+      it 'should be idempotent' do
+        expect(apply_manifest(
+          single_manifest, :catch_failures => true
+        ).exit_code).to be_zero
+      end
+    end
+
+    describe "secured REST endpoint" do
+      it 'denies unauthorized access' do
+        curl_with_retries(
+          'anonymous cluster health request',
+          default,
+          "-s -I -XGET http://localhost:9200/_cluster/health " \
+            "| grep '401 Unauthorized'", 0)
+      end
+
+      it 'permits authorized access' do
+        curl_with_retries(
+          'elastic user cluster health request',
+          default,
+          "-s -I -XGET -u #{@user}:#{@user_password} "\
+          "http://localhost:9200/_cluster/health " \
+            "| grep '200 OK'", 0)
+      end
+    end
+  end
+
+  describe 'changing passwords' do
+    describe 'password change manifest' do
+
+      let :passwd_manifest do
+        base_manifest + <<EOF
+elasticsearch::instance { ['es-01'] :  }
+
+Elasticsearch::Plugin { instances => ['es-01'],  }
+
+notify { 'change password' : } ~>
+elasticsearch::shield::user { '#{@user}pwchange':
+  password => '#{@user_password[0..5]}',
+  roles    => ['admin'],
+}
+EOF
+      end
+
+      it 'should apply cleanly' do
+        apply_manifest passwd_manifest, :catch_failures => true
+      end
+    end
+
+    describe "secured REST endpoint" do
+      it 'authorizes changed passwords' do
+        curl_with_retries(
+          'elastic user cluster health request with new password',
+          default,
+          "-s -I -XGET -u #{@user}pwchange:#{@user_password[0..5]} "\
+          "http://localhost:9200/_cluster/health " \
+            "| grep '200 OK'", 0)
```

Comment:
It looks like you could use a helper method around `curl_with_retries` (or maybe just curl). It's hard to see the intent of this test because of all the boilerplate you have to feed the function call.

Also, passing `| grep` as a part the `url` argument is a clever hack, but it's kinda nasty.